assets/js/workflow-diagram/WorkflowDiagram.tsx

```diff
@@ -224,6 +227,80 @@ export default function WorkflowDiagram(props: WorkflowDiagramProps) {
     }
   }, [workflow, flow, placeholders, el, isManualLayout, fixedPositions, selection]);
 
+  // This effect only runs when AI assistant visibility changes, not on every selection change
+  useEffect(() => {
+    if (!props.showAiAssistant) {
+      setDrawerWidth(0);
+      
+      // Fit view when AI assistant panel closes
```

Comment:
This should probably be a recenter rather than a fit. If the user is zoomed in then the view will reset and they'll lose context.

We could use a heuristic like "fit to all nodes that are currently visible". This is a trick we use elsewhere and will probably work well here